libjpeg-turbo/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+                       # via INCLUDE_DIRECTORIES property later on.
+                       SRCS dummy.c)
+
+# Determine compilation flags used for building Jpeg-turbo
+# Flags inherited from IDF build system and other IDF components:
+set(idf_include_directories $<TARGET_PROPERTY:idf::libjpeg-turbo,INCLUDE_DIRECTORIES>)
+set(includes "-I$<JOIN:${idf_include_directories}, -I>")
+
+
+if(CONFIG_COMPILER_OPTIMIZATION_DEFAULT)
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug)
+elseif(CONFIG_COMPILER_OPTIMIZATION_SIZE)
+    set(opt_args -DCMAKE_BUILD_TYPE=Relesee)
+elseif(CONFIG_COMPILER_OPTIMIZATION_PERF)
+    set(opt_args -DCMAKE_BUILD_TYPE=Relesee)
+elseif(CONFIG_COMPILER_OPTIMIZATION_NONE)
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug )
+else()
+    message(FATAL_ERROR "Unsupported optimization level")
+endif()
+
+set(extra_defines -D_DEFAULT_SOURCE)
+
+set(c_flags "${includes} ${extra_defines} ")
+set(cxx_flags "${includes} ${extra_defines} ")
+set(common_flags "-ggdb  -ffunction-sections -fdata-sections -fpermissive -lm")
+
+# set(COMPILE_FLAGS "-ggdb  -ffunction-sections -fdata-sections -fpermissive -lm")
+# set(COMPILE_FLAGS "-ggdb  -ffunction-sections -fdata-sections -fpermissive")
+
+if(CONFIG_IDF_TARGET_ARCH_XTENSA)
+    set(assert_flags "${assert_flags} -mlongcalls")
+endif()
+
+if(CONFIG_COMPILER_OPTIMIZATION_DEFAULT)
+    set(opt_flags "-Og ${common_flags} ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_C_FLAGS_RELEASE=${opt_flags}
+    -DCMAKE_CXX_FLAGS_RELEASE=${opt_flags})
+elseif(CONFIG_COMPILER_OPTIMIZATION_SIZE)
+    # We redefine the flags for Release instead.
+    set(opt_flags "-Os ${common_flags} ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Release
+                 -DCMAKE_C_FLAGS_RELEASE=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_RELEASE=${opt_flags})
+elseif(CONFIG_COMPILER_OPTIMIZATION_PERF)   
+    set(opt_flags "-O3 ${common_flags} ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Release
+                 -DCMAKE_C_FLAGS_RELEASE=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_RELEASE=${opt_flags})
+elseif(COMPILER_OPTIMIZATION_NONE)
+    set(opt_flags "-O0 ${common_flags} ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug
+                 -DCMAKE_C_FLAGS_DEBUG=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_DEBUG=${opt_flags})
+else()
+    set(opt_flags "-O0 ${common_flags} ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug
+                 -DCMAKE_C_FLAGS_DEBUG=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_DEBUG=${opt_flags})
+endif()
+
+idf_build_get_property(python PYTHON)
+
+include(ExternalProject)
+
+# Build jpeg-turbo in this directory:
+set(BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/libjpeg-build)
+
+# The list here just to make it common with other components
+set(jpegturbo_libraries
+    libjpeg
+)
+
+set(all_libraries)
+set(all_targets)
+
+foreach(libname ${jpegturbo_libraries})
```

Comment:
Why do we need foreach to loop over jpegturbo_libraries when jpegturbo_libraries only contains one library?